app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/provision_workflow/network.rb

```diff
@@ -31,7 +37,8 @@ def cloud_subnets(_options = {})
     return {} if zone.nil? || cloud_network.nil?
 
     method_log.debug("availability_zone value is #{zone} && cloud_network is #{cloud_network}")
-    subnets = ar_ems.cloud_subnets.select { |sn| sn[:availability_zone_id] == zone && sn.cloud_network.ems_ref == cloud_network }
+    all_cloud_subnets = resources_for_ui[:ems] ? ar_ems.cloud_subnets : ManageIQ::Providers::IbmCloud::VPC::NetworkManager::CloudSubnet.all
+    subnets = all_cloud_subnets.select { |sn| sn[:availability_zone_id] == zone && sn.cloud_network.ems_ref == cloud_network }
```

Comment:
I'm a little concerned about stuff like this though, have to be careful that this doesn't match on a subnet that is in a different EMS from the selected cloud_network if we only have the ems_ref to go on